pkg/controller/controller.go

```diff
@@ -173,8 +173,16 @@ func sendEvent(obj, oldObj interface{}, c *config.Config, notifiers []notify.Not
 		}
 	}
 
-	// Filter events
+	// Run custom filters
 	event = filterengine.DefaultFilterEngine.Run(obj, event)
+
+	// Filter events
+	// Skip older update events if timestamp difference is more than 30sec
+	if !event.TimeStamp.IsZero() && time.Now().Sub(event.TimeStamp) > 30*time.Second {
```

Comment:
This code is to be executed only for job update events..?
I guess we need to add the above specific conditon to avoid skipping of other events.

Reply:
This is not Job resources specific, this will be checked for each event

Reply:
@codenio We are comparing TimeStamp with time.Now(). Any live event won't have time difference more than few Microseconds. We just don't want to send a notification with having an older timestamp. We are not figuring out whether it is duplicate or not here

Reply:
@PrasadG193 thanks making suitable changes.